src/main/java/com/exadel/discount/controller/DiscountController.java

```diff
@@ -0,0 +1,57 @@
+package com.exadel.discount.controller;
+
+import com.exadel.discount.dto.discount.CreateDiscountDTO;
+import com.exadel.discount.dto.discount.DiscountDTO;
+import com.exadel.discount.service.DiscountService;
+import io.swagger.annotations.ApiOperation;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.util.List;
+import java.util.UUID;
+
+@Slf4j
+@Validated
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/discounts")
+public class DiscountController {
+
+    private final DiscountService discountService;
+
+    @GetMapping
+    @ApiOperation("Get all discounts")
+    public List<DiscountDTO> getAllDiscounts() {
+        return discountService.getAll();
+    }
+
+    @GetMapping("/{discountId}")
+    @ApiOperation("Get discount by ID")
+    public DiscountDTO getDiscountById(
+            @PathVariable(name = "discountId") @NotNull UUID id) {
+        return discountService.getById(id);
+    }
+
+    @PostMapping
+    @ApiOperation("Add new discount")
+    public DiscountDTO addDiscount(@RequestBody @Valid CreateDiscountDTO discountDTO) {
+        return discountService.save(discountDTO);
+    }
+
+    @DeleteMapping("/{id}")
+    @ApiOperation("Delete discount")
+    public void deleteDiscount(@PathVariable(name = "id") @NotNull UUID id) {
+        discountService.deleteById(id);
```

Comment:
NotFoundException with special messege will be good here
Or default can leave...